internal/gps/source.go

```diff
@@ -507,14 +507,16 @@ func (sg *sourceGateway) require(ctx context.Context, wanted sourceState) (errSt
 					}
 				}
 			case sourceHasLatestVersionList:
-				var pvl []PairedVersion
-				err = sg.suprvsr.do(ctx, sg.src.sourceType(), ctListVersions, func(ctx context.Context) error {
-					pvl, err = sg.src.listVersions(ctx)
-					return err
-				})
+				if len(sg.cache.getAllVersions()) == 0 {
```

Comment:
This other case doing a check threw me off:
```
case sourceExistsLocally:
 				if !sg.src.existsLocally(ctx) {
```
So I added that check thinking it could be the source of my problems, but never confirmed it. Happy to revert.